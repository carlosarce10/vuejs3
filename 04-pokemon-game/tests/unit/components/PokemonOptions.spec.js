import PokemonOptions from '@/components/PokemonOptions';
import { shallowMount } from '@vue/test-utils';
import { pokemons } from '../mocks/pokemos.mock';

describe('PokemonOptions component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons,
      },
    });
  });
  test('debe hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('debe mostrar las 4 opciones correctamente', () => {
    const pokemonList = wrapper.findAll('li');

    expect(pokemonList.length).toBe(4);
    expect(pokemonList[0].text()).toBe('bulbasaur');
    expect(pokemonList[1].text()).toBe('ivysaur');
    expect(pokemonList[2].text()).toBe('venusaur');
    expect(pokemonList[3].text()).toBe('charmander');
  });

  test('debe de emitir "selection" con sus respectivos parÃ¡metros al hacer click', () => {
    const [li1, li2, li3, li4] = wrapper.findAll('li');

    li1.trigger('click');
    li2.trigger('click');
    li3.trigger('click');
    li4.trigger('click');

    expect(wrapper.emitted('selection').length).toBe(4);
    expect(wrapper.emitted('selection')[0]).toEqual([1]);
    expect(wrapper.emitted('selection')[1]).toEqual([2]);
    expect(wrapper.emitted('selection')[2]).toEqual([3]);
    expect(wrapper.emitted('selection')[3]).toEqual([4]);
  });
});
