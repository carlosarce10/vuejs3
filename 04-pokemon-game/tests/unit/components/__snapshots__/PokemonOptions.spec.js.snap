// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions component debe hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul class="list-group">
    <li class="list-group-item">bulbasaur</li>
    <li class="list-group-item">ivysaur</li>
    <li class="list-group-item">venusaur</li>
    <li class="list-group-item">charmander</li>
  </ul>
</div>
`;
